### DISCUSSION TITLE
Option to stop seeing a message?

### DISCUSSION THREAD

**Discussion by peyre** [raceintospace/raceintospace#640] [Ideas]
I got to thinking today, about the warnings I've created for R&D and Purchasing, that experienced players might find them a nuisance sometimes.  I wonder if it would be a good idea to give an option to stop seeing a given message any more in the current game.  For instance a player flying Voskhod will start seeing this once it reaches 80% Safety.
![Voskhod](https://user-images.githubusercontent.com/3118014/166112937-294d5f52-ed8f-407e-8422-74efc64f97ec.png)

We could maybe do something like this, but the window doesn't default to Yes, so a player can't hit Enter to dismiss it like they can with the pop-up above.
![VoskhodYesNo](https://user-images.githubusercontent.com/3118014/166112975-206223ee-6f97-4436-b8e6-efd29522786e.png)

**Comment by rnyoakum** [comment]
I agree players should have a way to disable 'friendly help' messages. I was thinking of an option that players could select in the configuration file which would disable them, but your idea has its own merits. By current game did you mean the current RIS session, or something that would be preserved in the save file?

**Reply by peyre** [reply]
Just for the current game was what I was thinking.  There could be variables in Data for these.  A nonzero value wouldn't show a given warning, or something like that.

Maybe we could modify the Help function so that 0 means just an ok button (like now), 1 gives Yes and No, and 2 gives Yes and No _and_ accepts Enter as a Yes.  Or maybe 2 could give Continue and Cancel buttons, and accept Enter as continue.

**Reply by peyre** [reply]
Another idea might be to change the conditions for the warning so it doesn't show unless you're over Max R&D _and_ the Safety is above 85%, or what have you.  That would be helpful for Gemini/Voskhod, since you generally don't want to start your kicker etc. immediately once you reach Max R&D.

**Reply by rnyoakum** [reply]
I wouldn't mess with the current Help system if I could avoid it. If RIS has different requirements for a tutorial system, I would spin off a new Tutorial system similar to (and based on) the help system. 

You would also need to identify the space in Data to store the tutorial flags. At least until the next full version of RIS comes out and it can be updated to C++11...

**Reply by peyre** [reply]
Hm, ok.  Leaving the Help file alone is at least easy.  I can go with a Yes/No.  So how do I create a new item in Data to hold these toggles?

**Reply by rnyoakum** [reply]
The problem with editing data is that the Data structure is what is stored in save files. Currently, that means the Data portion of the save file is read, uncompressed, and copied _directly_ into the Data variable. There isn't any versioning system, yet, so modifying Data means there would be no backwards compatibility with  _any_ previously created save file. @hweimer [proposed](https://github.com/raceintospace/raceintospace/issues/21#issuecomment-680855532) using the Cereal library to fix this, but that requires updating RIS to C++11. We [were waiting](https://github.com/raceintospace/raceintospace/pull/478) until the next release before doing that.

So at present, the only way is to scavenge unused variables in Data. If you want to go that route, I may have a few ideas.
